### PR TITLE
add changes :mouse:

### PR DESCRIPTION
**Pull request by azat-co**
CC @AnnJohn @vicfriedman

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,145 @@
...
+Headers often contain authentication information as well. This enables user sessions.
+
+This lesson will cover how to read any the request headers and write any response headers on the server using the examples of Accept-Charset, and Content-Type, Content-Length headers.
+
+## Request Headers
+
+`request` object of the request handler has an object `headers`. It has names of the headers as keys/properties. The names are lower-cased. For example, if you want to get the value of the `Accept-Charset` request header, you can access `request.headers['accept-charset']`:
+
+```js
+console.log(request.headers['accept-charset'])
```
Comment (victhevenot) [inline README.md:26]: what info does the `accept-charset` property store?

File: README.md [hunk 2]
```diff
@@ -1 +1,145 @@
...
+
+```
+curl --header "Azat-Header: base64" localhost:3000
+```
+
+
+## response Headers
+
+When it comes to sending back the response, it's a good idea to write response headers. As mentioned before, clients will work more intelligently with out content when our servers provide proper response headers. For the same reason, we also want to send back the appropriate status code. 
+
```
Comment (victhevenot) [inline README.md:77]: students already know all about status codes - can delete this info

File: README.md [hunk 3]
```diff
@@ -1 +1,145 @@
...
+## Request Headers
+
+`request` object of the request handler has an object `headers`. It has names of the headers as keys/properties. The names are lower-cased. For example, if you want to get the value of the `Accept-Charset` request header, you can access `request.headers['accept-charset']`:
+
+```js
+console.log(request.headers['accept-charset'])
+```
+
+To send the request headers, we like to use curl:
```
Comment (victhevenot) [inline README.md]: why do we like to use curl? say something about how curl provides an easy way for us to test sending request headers. would it be possible to try it with a familiar URL (like google.com or something) or will that provide another layer of complexity?

File: README.md [hunk 4]
```diff
@@ -1 +1,145 @@
...
+When it comes to sending back the response, it's a good idea to write response headers. As mentioned before, clients will work more intelligently without content when our servers provide proper response headers. For the same reason, we also want to send back the appropriate status code. 
+
+There are four categories of status codes:
+
+* 2XX: for successfully processed requests
+* 3XX: for redirections or cache information
+* 4XX: for client-side errors
+* 5XX: for server-side errors
+
+Using the core `http` module in the request handler, we can write the header with `response.writeHead(statusCode[, statusMessage][, headers]))` where `headers` is an object. For example, if we want to you send back status 200 with the message okay and two headers `Content-Type` and `Content-Length`, then we can use these statements:
```
Comment (victhevenot) [inline README.md]: can we write `response.writeHead(statusCode[, statusMessage][, headers]))` in a different way, the brackets are confusing..the status message in the example below isn't in an array, and neither are the properties and values.

as a suggestion: 
`response.writeHead(statusCode, statusMessage,  { headers }`

**Comment by vicfriedman** [comment]
need to update the intro to do more than just list the skills students are practicing

File: README.md (hunk 4 above)
Comment (azat-co) [inline README.md]: `[]` means optional. I added more instructions. I'm surprised students don't know this already. MDN uses it a lot as well as almost all official documentation.

**Pull request merged**